packages/wifi_basic/lib/wifi_basic.dart

```diff
@@ -42,21 +16,40 @@ class WiFiBasic {
   final _isSupportedMemo = AsyncMemoizer<bool>();
   final _getGenerationMemo = AsyncMemoizer<WiFiGenerations>();
 
-  Future<bool> isSupported() => _isSupportedMemo
-      .runOnce(() async => await _channel.invokeMethod('isSupported'));
-
-  Future<WiFiGenerations> getGeneration() => _getGenerationMemo.runOnce(
-      () async => (await _channel.invokeMethod("getGeneration") as int?)
-          .toWifiGeneration());
-
-  Future<bool> isEnabled() async => await _channel.invokeMethod('isEnabled');
-
-  Future<bool> setEnabled(bool enabled) async =>
-      await _channel.invokeMethod('setEnabled', {"enabled": enabled});
-
-  Future<void> openSettings() async =>
-      await _channel.invokeMethod("openSettings");
-
-  Future<WiFiInfo> getCurrentInfo() async =>
-      WiFiInfo._fromMap(await _channel.invokeMapMethod("getCurrentInfo") ?? {});
+  Future<bool> isSupported() {
+    return _isSupportedMemo.runOnce(() async {
```

Comment:
instead of memoizer can we do it with simple var? if null we calculate else return the result directly. we can avoid depency for `async` in that case.

Reply:
I do think so. At least I don't see any problems this could cause for now.